lib/Model/IMAPMessage.php

```diff
@@ -630,6 +645,9 @@ public function toDbMessage(int $mailboxId): \OCA\Mail\Db\Message {
 
 		$msg->setUid($this->getUid());
 		$msg->setMessageId($this->getMessageId());
+		$msg->setReferences($this->getRawReferences());
```

Comment:
I'm just thinking here if it would make sense to split them already and just store it as json or something? Basically process on input. So that you don't have to do it every time afterwards? But just a suggestion.

Reply:
yep, I was also thinking about that

Reply:
Let's just do that

Reply:
done

Reply:
But this is the other way around right? Now you still store them flat and process on setting it?

Reply:
They are parsed and set *once* during the insert. What would you like to see different? The layer at which the parsing happens?

Reply:
ah I'm just stupid then